lib/MusicBrainz/Server/Data/Relationship.pm

```diff
@@ -230,6 +230,55 @@ sub _load_related_info {
     $self->load_entities(@rels);
 }
 
+sub load_credits
+{
+    my ($self, $type, @ids) = @_;
+    my @types = map { [ sort($type, $_) ] } @RELATABLE_ENTITIES;
+    my @credits;
+    foreach my $t (@types) {
+        my $type0 = $t->[0];
+        my $type1 = $t->[1];
+        my $query;
+        my @params;
+
+        my $type0_query = <<~"SQL";
+            SELECT entity0_credit
+              FROM l_${type0}_${type1}
+             WHERE entity0 = any(?)
+               AND entity0_credit != ''
+            SQL
+
+        my $type1_query = <<~"SQL";
+            SELECT entity1_credit
+              FROM l_${type0}_${type1}
+             WHERE entity1 = any(?)
+               AND entity1_credit != ''
+            SQL
+
+        if ($type eq $type0 && $type eq $type1) {
+            $query = <<~"SQL";
+                (
+                    $type0_query
+                ) UNION (
+                    $type1_query                    
+                )
+                SQL
+            push @params, @ids, @ids;
+        } elsif ($type eq $type0) {
+            $query = $type0_query;
+            push @params, @ids;
+        } elsif ($type eq $type1) {
+            $query = $type1_query;
+            push @params, @ids;
+        }
+
+        my $results = $self->sql->select_single_column_array($query, @params);
+        push @credits, @$results;
+    }
+    my @sorted_credits = uniq sort {$a cmp $b} @credits;
+    return @sorted_credits;
```

Comment:
I'd suggest returning an array ref, which I've read is more efficient for large lists.
```suggestion
    return \@sorted_credits;
```